src/Multicall3NoCheck.sol

```diff
@@ -0,0 +1,85 @@
+/**
+ * Submitted for verification at Etherscan.io on 2022-05-17
+ */
+
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import "forge-std/console.sol";
+
+/// @title Multicall3
+/// @notice Aggregate results from multiple function calls
+/// @dev Multicall & Multicall2 backwards-compatible
+/// @dev Aggregate methods are marked `payable` to save 24 gas per call
+/// @author Michael Elliot <mike@makerdao.com>
+/// @author Joshua Levine <joshua@makerdao.com>
+/// @author Nick Johnson <arachnid@notdot.net>
+/// @author Andreas Bigger <andreas@nascent.xyz>
+/// @author Matt Solomon <matt@mattsolomon.dev>
+/// Multicall3 contract without msg.value == valAccumulator check
+/// in aggregate3Value function
+contract Multicall3NoCheck {
+    struct Call3Value {
+        address target;
+        bool allowFailure;
+        uint256 value;
+        bytes callData;
+    }
+
+    struct Result {
+        bool success;
+        bytes returnData;
+    }
+
+    address public immutable thisMulticall3;
+
+    constructor() {
+        thisMulticall3 = address(this);
+    }
+
+    /// @notice Aggregate calls with a msg value
+    /// @notice Reverts if msg.value is less than the sum of the call values
+    /// @param calls An array of Call3Value structs
+    /// @return returnData An array of Result structs
+    function aggregate3Value(Call3Value[] calldata calls) public payable returns (Result[] memory returnData) {
```

Comment:
we should port over the corresponding multicall3 tests into `test/multicall/Multicall3NoValueCheck.t.sol`